app/graphql/types/profile.rb

```diff
@@ -11,7 +11,27 @@ class Profile < Types::BaseObject
     field :description, String, null: true
     field :ref_id, String, null: false
     field :compliance_threshold, Float, null: false
-    field :rules, [::Types::Rule], null: true
+    field :rules, [::Types::Rule], null: true do
+      argument :identifier, String,
+               'Rule identifier to filter by', required: false
+      argument :references, [String],
+               'Rule references to filter by', required: false
+    end
+    def rules(args = {})
+      rules = object.rules
+
+      if args.dig(:identifier)
```

Comment:
Seems like these 2 could be moved to a helper method? It's nearly identical code

Reply:
Also it's separate to the rest of the methods down below